http.go

```diff
@@ -26,6 +28,11 @@ type ErrorType struct {
 
 // WriteJSONResponse write to te response stream
 func WriteJSONResponse(w http.ResponseWriter, response *Response) {
+	for header, values := range response.Headers {
+		for i := range values {
+			w.Header().Add(header, values[i])
```

Comment:
Array indirection may be expensive.
`for _, v := range values` should be quite faster.

Reply:
Also, saving the repeated function call for w.Header() should squeeze a bit of extra performance.